cobbler/items/abstract/inheritable_item.py

```diff
@@ -0,0 +1,361 @@
+"""
+"InheritableItem" the entry point for items that have logical parents and children.
+
+Changelog:
+    * V3.4.0 (unreleased):
+        * Initial creation of the class
+"""
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Enno Gotthold <enno.gotthold@suse.com>
+
+from abc import ABC
+from typing import TYPE_CHECKING, Any, Dict, List, NamedTuple, Optional, Union
+
+from cobbler.cexceptions import CX
+from cobbler.decorator import LazyProperty
+from cobbler.items.abstract.base_item import BaseItem
+
+if TYPE_CHECKING:
+    from cobbler.api import CobblerAPI
+    from cobbler.items.distro import Distro
+    from cobbler.items.menu import Menu
+    from cobbler.items.profile import Profile
+    from cobbler.items.system import System
+    from cobbler.settings import Settings
+
+
+class HierarchyItem(NamedTuple):
+    """
+    NamedTuple to display the dependency that a single item has.
```

Comment:
WDYT, would it be helpful to expand the doc string a little? 

```suggestion
    NamedTuple to display the dependency that a single item has.
    The `dependant_item_type` object is stored in the `dependant_type_attribute` attribute of the Item.
    
    For example, an item with HierarchyItem("profile", "repos") contains `Profile` objects in the `repos` attribute.
```